src/main/kotlin/com/cognifide/gradle/aem/instance/Instance.kt

```diff
@@ -62,24 +59,65 @@ interface Instance : Serializable {
             }
         }
 
-        fun properties(project: Project): List<RemoteInstance> {
-            return project.properties.filterKeys { Patterns.wildcard(it, REMOTE_PROP_PATTERN) }.map {
-                val nameParts = StringUtils.substringBetween(it.key, ".remote.", ".httpUrl")!!.split("-")
+        fun properties(project: Project): List<Instance> {
+            val localInstances = collectProperties(project, "local").map {
+                val (name, props) = it
+                val nameParts = name.split("-")
                 if (nameParts.size != 2) {
-                    throw InstanceException("Instance list property '${it.key}' does not have valid name part.")
+                    throw InstanceException("Local instance name has invalid format: '$name'.")
                 }
-
                 val (environment, typeName) = nameParts
-                val httpUrl = it.value as String?
-                if (httpUrl.isNullOrBlank()) {
-                    throw InstanceException("Instance list property '${it.key}' value cannot be blank.")
+                val httpUrl = props["httpUrl"]
+                        ?: throw InstanceException("Local instance named '$name' must have property 'httpUrl' defined.")
+
+                LocalInstance.create(httpUrl, {
+                    this.environment = environment
+                    this.typeName = typeName
+
+                    props["password"]?.let { this.password = it }
```

Comment:
love this lines, how much boring complexity covered by Kotlin / comparing to Java